assets/scripts/dialogs/UpgradeDialog.jsx

```diff
@@ -1,56 +1,115 @@
-    console.log('onToken', token, addresses)
+  constructor (props) {
+    super(props)
+
+    this.state = {
+      loading: false,
+      error: null,
+      data: null
+    }
+  }
+
+  onToken = async (token) => {
+    const { userId } = this.props
+    const requestBody = { userId, token }
+
+    this.setState({ loading: true })
+    try {
+      const { data } = await axios.post(API_URL + 'v1/pay', requestBody)
+      this.setState({ loading: false, data })
+    } catch (err) {
+      this.setState({ loading: false, error: err })
+    }
   }
 
   render () {
+    const { intl } = this.props
+    const { data, loading } = this.state
+    const stripeName = intl.formatMessage({
+      id: 'upgrade.name',
+      defaultMessage: 'Streetmix Enthusiast Plan'
+    })
+    const stripeDescription = intl.formatMessage({
+      id: 'upgrade.description',
+      defaultMessage: 'stripe description goes here'
+    })
+    const stripeLabel = intl.formatMessage({
+      id: 'upgrade.label',
+      defaultMessage: 'Subscribe'
+    })
+
+    let activePanel
+    if (loading) {
+      activePanel = (
+        <p>
+          <FormattedMessage id="upgrade.loading" defaultMessage="Loading..." />
+        </p>)
+    } else if (data) {
+      activePanel = (
+        <p>
+          <FormattedMessage id="upgrade.success" defaultMessage="Thank you!" />
+        </p>)
+    } else {
+      activePanel = (<React.Fragment>
+        <p>
+          <FormattedMessage id="upgrade.body" defaultMessage={DEFAULT_BODY} />
+        </p>
+        <StripeCheckout
+          amount={500}
+          name={stripeName}
+          description={stripeDescription}
+          label={stripeLabel}
+          locale="auto"
+          stripeKey={STRIPE_API_KEY}
+          token={this.onToken}
+          zipCode
+        />
+      </React.Fragment>)
+    }
+
     return (
       <Dialog>
         {(closeDialog) => (
           <div className="upgrade-dialog" dir="ltr">
             <header>
               <h1>
-                Upgrade
+                <FormattedMessage id="upgrade.title" defaultMessage="Upgrade" />
               </h1>
             </header>
-            <div className="dialog-content">
-              <p>
-                <strong>TEST CONTENT!</strong>
-              </p>
-              <form action="/pay" method="post">
-                <article>
-                  <label>Amount: $5.00</label>
-                </article>
-                <StripeCheckout
-                  amount={1000}
-                  name="Streetmix"
-                  description="Activate your Enthusiast subscription"
-                  locale="auto"
-                  stripeKey={STRIPE_API_KEY}
-                  token={this.onToken}
-                  zipCode
-                />
-              </form>
-            </div>
+            {activePanel}
           </div>
         )}
       </Dialog>
     )
   }
 }
 
-export default UpgradeDialog
+UpgradeDialog.propTypes = {
+  intl: intlShape.isRequired,
```

Comment:
Although I prefer moving to a function component, `intlShape` should no longer be used because it's not available in `react-intl@3.x.x`.